### PR TITLE
Event list rework issue135

### PR DESCRIPTION
**Pull request by MayJue**
This PR fixes issue #135. 

We remove the accordion on the list of events on the home page and replace them with tables(except for one. 
Then added a note for when there are no events. 
Also created test.

to test, go to the home page and select the different tabs.

### CODE REVIEW

File: app/logic/events.py [hunk 1]
```diff
@@ -95,50 +95,47 @@ def saveEventToDb(newEventData):
...
-
-    for event in eventQuery.objects():
-        programs.setdefault(Program.get_by_id(event.program_id), []).append(event)
-
-    return programs
-
-def groupEventsByCategory(term):
-
-    term = Term.get_by_id(term)
+def getStudentLeadProgram(term):
```
Comment (BrianRamsay) [inline app/logic/events.py]: This name should be StudentLed, not StudentLead

File: app/logic/events.py [hunk 2]
```diff
@@ -95,50 +95,47 @@ def saveEventToDb(newEventData):
...
 
     studentLedEvents = (Event.select(Event, Program.id.alias("program_id"))
                              .join(ProgramEvent)
                              .join(Program)
                              .where(Program.isStudentLed,
+                                    Event.isTraining == False,
```
Comment (BrianRamsay) [inline app/logic/events.py]: Is there a reason we are excluding training events from this list?

File: app/templates/events/event_list.html [hunk 3]
```diff
@@ -51,77 +51,123 @@ <h4><strong>Term</strong></h4>
...
+                    <th scope="col">Time</th>
+                    <th scope="col">Location</th>
+                    <th scope="col">Invitation</th>
+                  </tr>
+                </thead>
+                <tbody>
+
+                    {% for event in events %}
+                      <tr>
+                        <td> <a href="/{{program.id}}/{{event.id}}/edit_event" class="link-primary">{{event.name}}</a></td>
```
Comment (BrianRamsay) [inline app/templates/events/event_list.html]: This isn't the right URL anymore. program is unnecessary when editing an event

File: app/templates/events/event_list.html [hunk 4]
```diff
@@ -51,77 +51,123 @@ <h4><strong>Term</strong></h4>
...
+        <th scope="col">Time</th>
+        <th scope="col">Location</th>
+        <th scope="col">Invitation</th>
+      </tr>
+    </thead>
+    <tbody>
+
+        {% for event in events %}
+          <tr>
+            <td> <a href="/{{event.program_id}}/{{event.id}}/edit_event" class="link-primary">{{event.name}}</a></td>
```
Comment (BrianRamsay) [inline app/templates/events/event_list.html]: This url is wrong. Make sure when you're replacing a big block, that you identify changes that were made in that block

File: app/templates/events/event_list.html [hunk 5]
```diff
@@ -51,77 +51,123 @@ <h4><strong>Term</strong></h4>
...
+            <td>{{event.startDate.strftime('%m-%d-%Y')}}</td>
+            <td>{{event.timeStart.strftime('%I:%M %p')}}</td>
+            <td>{{event.location}}</td>
+            {% if (user.isCeltsAdmin) or (user.isCeltsStudentStaff) %}
```
Comment (BrianRamsay) [inline app/templates/events/event_list.html]: use the `user.isAdmin` shortcut

File: app/templates/events/event_list.html [hunk 6]
```diff
@@ -51,77 +51,123 @@ <h4><strong>Term</strong></h4>
...
+              <td><i class="bi bi-envelope-fill"> Email</i></td>
+              {# FIXME: the Email icon should be clickable and lead to email template #}
+            {% else %}
+              {% if (event.event.startDate < currentTime.date()) or (event.event.timeStart < currentTime.time() and event.event.startDate == currentTime.date())%}
```
Comment (BrianRamsay) [inline app/templates/events/event_list.html]: we'll use the `event.isPast` shortcut if it's available by the time this PR is ready. we'll see.

File: tests/code/test_event_list.py [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+                                timeStart = "18:00:00",
+                                timeEnd = "21:00:00",
+                                location = "basement",
+                                isRecurring = 0,
+                                isRsvpRequired = 0,
+                                isTraining = 0,
+                                isService = 0,
+                                startDate = 2021-12-12,
+                                endDate = 2021-12-13)
+        studentProgramEvent = ProgramEvent.create(program = 2, event = studentLead)
```
Comment (BrianRamsay) [inline tests/code/test_event_list.py]: if you want to save some space, you don't need to enter values for fields with defaults (like all the boolean ones that are 0)

**Review by BrianRamsay — CHANGES_REQUESTED** [review]

**Comment by tylerpar99** [comment]
@BrianRamsay gets 6.5% credit for this PR per our contractual agreement.

File: app/templates/events/event_list.html [hunk 8]
```diff
@@ -51,77 +51,123 @@ <h4><strong>Term</strong></h4>
...
+        <th scope="col">Time</th>
+        <th scope="col">Location</th>
+        <th scope="col">Invitation</th>
+      </tr>
+    </thead>
+    <tbody>
+
+        {% for event in events %}
+          <tr>
+            <td> <a href="/{{event.id}}/edit_event" class="link-primary">{{event.name}}</a></td>
```
Comment (BrianRamsay) [inline app/templates/events/event_list.html]: URL is not correct (see the removed code above)

File: tests/code/test_event_list.py [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+        assert True not in [event.name == "Test Student Lead" for event in trainingProgram]
+        assert True not in [event.name == "Test Training Program" for event in trainingProgram2]
+
+        assert bonnerProgram
+        assert True in [event.name == "Test Bonner Program" for event in bonnerProgram]
+        assert True not in [event.name == "Test Student Lead" for event in bonnerProgram]
+
+        assert oneTimeEvents
+        assert True in [event.name == "Test One Time" for event in oneTimeEvents]
+        assert True not in [event.name == "Test Student Lead" for event in oneTimeEvents]
```
Comment (BrianRamsay) [inline tests/code/test_event_list.py]: For these asserts, you have an event object you can use to directly check for inclusion (or not) in the event list.

**Review by BrianRamsay — CHANGES_REQUESTED** [review]
test_event_list.py has Windows line endings

File: app/templates/events/event_list.html [hunk 10]
```diff
@@ -51,77 +51,123 @@ <h4><strong>Term</strong></h4>
...
+                </thead>
+                <tbody>
+
+                    {% for event in events %}
+                      <tr>
+                        <td> <a href="/event/{{event.id}}/edit" class="link-primary">{{event.name}}</a></td>
+                        <td>{{event.startDate.strftime('%m-%d-%Y')}}</td>
+                        <td>{{event.timeStart.strftime('%I:%M %p')}}</td>
+                        <td>{{event.location}}</td>
+                        {% if (user.isCeltsAdmin) or (user.isCeltsStudentStaff) %}
```
Comment (BrianRamsay) [inline app/templates/events/event_list.html]: Here's another place you could use `user.isAdmin`

**Review by BrianRamsay — CHANGES_REQUESTED** [review]
Go ahead and change the tab bar to tabs instead of pills (or whatever they are called)

**Review by BrianRamsay — APPROVED** [review]
Good work!

**Pull request merged**